### PR TITLE
[Group] Add group ID retrieval

### PR DESCRIPTION
**Pull request by jepenven-silabs**
#### Problem
While Node ID can be retrieved from the message header, the code for the Group ID retrieval was missing

#### Change overview
Added code to handle the Group ID in the message header

#### Testing
How was this tested? (at least one bullet point required)
* I If unit tests were added, how do they cover this issue?
Same coverage as for the Node ID inside the TestMessagesHeader unit test with Set/Clear/Encode/Decode

### CODE REVIEW

File: src/lib/core/GroupId.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+namespace chip {
+
+// TODO: Consider making this a class and the various utility methods static
+// methods.
+using GroupId = uint16_t;
+
+constexpr GroupId kUndefinedGroupId = 0;
+
+constexpr GroupId kMinUniversalGroupId = 0x8000;
+constexpr GroupId kMaxUniversalGroupId = 0xFFFF;
```
Comment (msandstedt) [inline src/lib/core/GroupId.h:31]: Would it be preferable to explicitly define the reserved range (0x8000 - 0xFFFC), all proxies (0xFFFD), all sleep devices (0xFFFE) and all nodes (0xFFFF) codes?

**Review by woody-apple — APPROVED** [review]

File: src/app/util/basic-types.h [hunk 2]
```diff
@@ -27,6 +27,8 @@
 
 // Pull in NodeId
 #include <lib/core/NodeId.h>
+// Pull in GroupId
```
Comment (andy31415) [inline src/app/util/basic-types.h:30]: do these comments (this and above) provide any value?  They seem to say:

```cpp
// Pull in Foo
#include <.../Foo.h>
```

File: src/lib/core/GroupId.h [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <cstdint>
+
+namespace chip {
+
+// TODO: Consider making this a class and the various utility methods static
```
Comment (andy31415) [inline src/lib/core/GroupId.h:24]: I would remove this todo for now - it we really think this is required, we should do it now. However I believe existing code is probably good enough.

**Review by andy31415 — APPROVED** [review]

File: src/app/util/basic-types.h (hunk 2 above)
Comment (jepenven-silabs) [reply src/app/util/basic-types.h:30]: Removed

File: src/lib/core/GroupId.h (hunk 1 above)
Comment (jepenven-silabs) [reply src/lib/core/GroupId.h:29]: Sure, added

**Review by msandstedt — APPROVED** [review]

**Pull request merged**